server/src/main/java/org/elasticsearch/index/mapper/DocumentParser.java

```diff
@@ -941,7 +941,9 @@ public Query termQuery(Object value, SearchExecutionContext context) {
             protected void parseCreateField(DocumentParserContext context) {
                 // Run-time fields are mapped to this mapper, so it needs to handle storing values for use in synthetic source.
                 // #parseValue calls this method once the run-time field is created.
-                if (context.dynamic() == ObjectMapper.Dynamic.RUNTIME && context.canAddIgnoredField()) {
+                var fieldType = context.mappingLookup().getFieldType(path);
```

Comment:
This is another bug. The value of mapped runtime fields wasn't added to ignored source.  I think we can also include this change in this PR. Otherwise some assertion in added PR need to be removed.